relays/substrate/src/headers_target.rs

```diff
@@ -88,10 +88,14 @@ where
 		let data = Bytes(Vec::new());
 
 		let encoded_response = self.client.state_call(call, data, None).await?;
-		let decoded_response: (P::Number, P::Hash) =
+		let decoded_response: Vec<(P::Number, P::Hash)> =
 			Decode::decode(&mut &encoded_response.0[..]).map_err(SubstrateError::ResponseParseFailed)?;
 
-		let best_header_id = HeaderId(decoded_response.0, decoded_response.1);
+		let best_header = decoded_response.last().expect(
```

Comment:
I don't think this expect conforms to our expect/unwrap rules :) You're in relay code and get data from external component (node -> runtime pallet). There's a lot of components that may fail and it'll lead to empty vector here. So this expect is the same as using `Decode::decode().expect("pallet properly encodes hashes and numbers; qed")` couple of lines above. I'd return some error here instead. But it's just a mumble

Reply:
I did have second thoughts when `expect()`-ing yesterday, but figured it was fine. You're right though, we shouldn't be relying on cross-component assumptions like that, so I've changed it to an error. I _might_ be abusing `codec::Error` a little, but I didn't want to introduce a whole new error variant for this